Mathlib/Topology/ContinuousMap/Algebra.lean

```diff
@@ -527,18 +527,25 @@ instance instSMul [SMul R M] [ContinuousConstSMul R M] : SMul R C(α, M) :=
   ⟨fun r f => ⟨r • ⇑f, f.continuous.const_smul r⟩⟩
 
 @[to_additive]
-instance [LocallyCompactSpace α] [SMul R M] [ContinuousConstSMul R M] :
-    ContinuousConstSMul R C(α, M) :=
-  ⟨fun γ => continuous_of_continuous_uncurry _ (continuous_eval.const_smul γ)⟩
+instance [SMul R M] [ContinuousConstSMul R M] : ContinuousConstSMul R C(α, M) where
+  continuous_const_smul r := continuous_postcomp ⟨_, continuous_const_smul r⟩
 
 @[to_additive]
-instance [LocallyCompactSpace α] [TopologicalSpace R] [SMul R M] [ContinuousSMul R M] :
-    ContinuousSMul R C(α, M) :=
-  ⟨by
-    refine continuous_of_continuous_uncurry _ ?_
-    have h : Continuous fun x : (R × C(α, M)) × α => x.fst.snd x.snd :=
-      continuous_eval.comp (continuous_snd.prodMap continuous_id)
-    exact (continuous_fst.comp continuous_fst).smul h⟩
+instance [TopologicalSpace R] [SMul R M] [ContinuousSMul R M] :
+    ContinuousSMul R C(α, M) := by
```

Comment:
Am I right that this can be generalized to something like this: if `g : Y × Z -> Z` is continuous, then postcomposition with `g(c, -)` is continuous as a function on `Y × X`? Is there a `Tendsto` version? Could you please add them to the `CompactOpen` file?

Reply:
I think the proof can be generalized to `X × Y -> Z` is continuous implies `X × C(α, Y) -> C(α, Z)` continuous. Is this what you want or is there something more general? As for the `Tendsto`, the best I could get is something like `(f : C(X, Y)) : Filter.RTendsto (α := C(X, Y)) (fun g x ↦ x ∈ g '' K) (𝓝 f) (𝓝ˢ (f '' K))` for compact `K` which doesn't seem very useful.

Reply:
I think I have identified the key, which is the map from `X × C(Y, Z)` to `C(Y, X × Z)` is continuous. This seems to be a more general version of `ContinuousMap.continuous_coev`.

I am still not sure about what a `Tendsto` version would look like though.